src/core.jsx

```diff
@@ -22,41 +22,36 @@ const getDisplayName = c => c.displayName || c.name || 'Component'
 export default (
   {
     LoadingIndicator,
+    ErrorIndicator,
     print,
     load,
+    error,
   } = {},
 ) => {
   const loadFunctionName = isString(load) ? load : 'load'
-  const isPrintArray = Array.isArray(print)
-  const isPrintFunction = isFunction(print)
   const isLoadFunction = isFunction(load)
 
-  const isLoaded = (props, state, context) => {
-    // Print is undefined,
-    // we rely on 'props.loaded' if present
-    // if not, we directly print the component
-    if (print === undefined) {
-      const { loaded } = props
-      return loaded === undefined ? true : !!loaded
+  const hasStatus = (prop, propProcessor, defaultProp, defaultValue) => (props, state, context) => {
+    if (prop === undefined) {
+      const status = props[defaultProp]
+      return status === undefined ? defaultValue : !!status
     }
 
-    // Print is an array
-    // Implicitly meaning that this is an array of props
-    if (isPrintArray) {
-      return print
-        .map(p => Boolean(props[p]))
-        .reduce((allProps, currentProp) => allProps && currentProp)
+    if (Array.isArray(prop)) {
+      const boolProps = prop.map(p => Boolean(props[p]))
```

Comment:
We should avoir this to kind of writing (I know this was the case before, you can open an issue if you want to argue).
But here in `hasStatus` we have `!!` technique and `Boolean` technique